testing/test_hookcaller.py

```diff
@@ -213,3 +215,14 @@ def hello(self, arg):
     assert not hasattr(hook, "world")
     pm.unregister(plugin)
     assert hook.hello(arg=3) == []
+
+    # the `specname` argument overrides the function name when registering a hook caller
+    class Plugin2(object):
+        @hookimpl(specname="hello")
+        def foo(self, arg):
+            return arg + 1
+
+    plugin = Plugin2()
+    pm.register(plugin)
+    out = hook.hello(arg=3)
+    assert out == [4]
```

Comment:
this feels like it should be a separate test

Reply:
thanks, pulled it out into a new one